saleor/graphql/shop/mutations.py

```diff
@@ -397,18 +402,23 @@ def perform_mutation(cls, _root, info, **data):
             "automatically_confirm_all_new_orders",
             "automatically_fulfill_non_shippable_gift_card",
         ]
-        site = get_site_promise(info.context).get()
-        instance = site.settings
-        update_fields = []
+        update_fields = {}
+        channels = channel_models.Channel.objects.all()
         for field in FIELDS:
             value = data["input"].get(field)
             if value is not None:
-                setattr(instance, field, value)
-                update_fields.append(field)
+                for channel in channels:
+                    setattr(channel, field, value)
+                update_fields[field] = value
 
         if update_fields:
-            instance.save(update_fields=update_fields)
-        return OrderSettingsUpdate(order_settings=instance)
+            channel_models.Channel.objects.bulk_update(
+                channels,
+                list(update_fields.keys()),
+            )
+        for field in FIELDS:
+            update_fields.setdefault(field, True)
```

Comment:
I don't think that this will be truth for all cases. If you update single value, the second one could be set to `False`.
I would propose rebuild `perform_mutation` to something like this one:
```    @classmethod
    def perform_mutation(cls, _root, info, **data):
        fields_to_update = {}
        if "automatically_confirm_all_new_orders" in data:
            fields_to_update["automatically_confirm_all_new_orders"] = data["automatically_confirm_all_new_orders"]
        if "automatically_fulfill_non_shippable_gift_card" in data:
            fields_to_update["automatically_fulfill_non_shippable_gift_card"] = data["automatically_fulfill_non_shippable_gift_card"]
        Channel.objects.update(**fields_to_update)
        first_channel = Channel.objects.order_by("slug").first()

        return OrderSettingsUpdate(order_settings=<<values-from-first-channel>>)
```
Also, would be worth to mention in the `order_settings = graphene.Field(OrderSettings, description="Order settings.")` (line 383), that the field will return the order settings for the first channel in alpahbetical order for slug (same approach as we have for `orderSettings` query)